### PR TITLE
Fix tracing so handler errors are tracked

### PR DESCRIPTION
**Pull request by DarcyRaynerDD**
### What does this PR do?

Fixes bug where root span of trace wouldn't be marked as an error when an exception is thrown.

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -112,34 +112,30 @@ export function datadog<TEvent, TResult>(
...
           localResult = (await promHandler(localEvent, localContext)) as TResult | undefined;
         } catch (err) {
           if (finalConfig.enhancedMetrics) {
             incrementErrorsMetric(metricsListener, localContext);
           }
-          localError = err;
-          localDidError = true;
+          throw err;
+        } finally {
+          await metricsListener.onCompleteInvocation();
```
Comment (Czechh) [inline src/index.ts:131]: Can we test that this function is called independently of a rejection from `promHandler`. Ideally a test that failed before and now passes to improve confidence in the change.

Comment (DarcyRaynerDD) [reply src/index.ts:131]: Good point, I'll add another test case

**Review by Czechh — APPROVED** [review]
sorry i meant a unit, but this also works 👍

**Comment by DarcyRaynerDD** [comment]
@Czechh Ahh, yeah. I should have mentioned, I think we already have a test case making sure everything get's called during cleanup when there is an error in the handler. This logic change shouldn't alter that behavior. The actual change was to make sure an exception was thrown through the tracer.wrapped function, (before we were returning a struct with an error or the result from the handler). This insures the error is attached to the trace. I added the integration tests, because it's the only way to verify the change in the trace output.

**Pull request merged**